bundles/org.openhab.binding.dirigera/src/test/java/org/openhab/binding/dirigera/internal/handler/DirigeraBridgeProvider.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dirigera.internal.handler;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import static org.openhab.binding.dirigera.internal.Constants.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.openhab.binding.dirigera.internal.mock.CallbackMock;
+import org.openhab.binding.dirigera.internal.mock.DicoveryServiceMock;
+import org.openhab.binding.dirigera.internal.mock.DirigeraAPISimu;
+import org.openhab.binding.dirigera.internal.mock.DirigeraHandlerManipulator;
+import org.openhab.binding.dirigera.internal.mock.HandlerFactoryMock;
+import org.openhab.core.storage.Storage;
+import org.openhab.core.storage.StorageService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.internal.BridgeImpl;
+import org.openhab.core.thing.internal.ThingImpl;
+
+/**
+ * {@link DirigeraBridgeProvider} Tests device handler creation, initializing and refresh of channels
+ *
+ * @author Bernd Weymann - Initial Contribution
+ */
+@NonNullByDefault
+public class DirigeraBridgeProvider {
+    public static Bridge prepareSimuBridge() {
+        return prepareSimuBridge("src/test/resources/home/home.json", false, List.of());
+    }
+
+    /**
+     * Prepare bridge which can be used with DirigraAPISimu Provider
+     *
+     * @return Bridge
+     */
+    public static Bridge prepareSimuBridge(String homeFile, boolean discovery, List<String> knownDevicesd) {
+        String ipAddress = "1.2.3.4";
+        HttpClient httpMock = mock(HttpClient.class);
+        DirigeraAPISimu.fileName = homeFile;
+        /**
+         * Prepare persistence
+         */
+        // prepare persistence data
+        JSONObject storageObject = new JSONObject();
+        JSONArray knownDevices = new JSONArray(knownDevicesd);
+        knownDevices.put("594197c3-23c9-4dc7-a6ca-1fe6a8455d29_1");
+        storageObject.put(PROPERTY_DEVICES, knownDevices.toString());
+        storageObject.put(PROPERTY_TOKEN, "unit-test");
+        // now mock it
+        Storage<String> mockStorage = mock(Storage.class);
```

Comment:
This gives a warning, can you fix it.

Reply:
Using now `org.openhab.core.test.storage.VolatileStorageService`